pkg/workload/workload.go

```diff
@@ -236,10 +243,63 @@ func BaseSSAWorkload(w *kueue.Workload) *kueue.Workload {
 	return wlCopy
 }
 
-// AdmissionPatch creates a new object based on the input workload that
-// contains the admission. The object can be used in Server-Side-Apply.
-func AdmissionPatch(w *kueue.Workload) *kueue.Workload {
+// SetAdmission applies the provided admission to the workload.
+// The WorkloadAdmitted and WorkloadEvicted are added or updated if necessary.
+func SetAdmission(w *kueue.Workload, admission *kueue.Admission) {
+	w.Status.Admission = admission
+	admittedCond := metav1.Condition{
+		Type:               kueue.WorkloadAdmitted,
+		Status:             metav1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+		Reason:             "Admitted",
+		Message:            fmt.Sprintf("Admitted by ClusterQueue %s", w.Status.Admission.ClusterQueue),
+	}
+	apimeta.SetStatusCondition(&w.Status.Conditions, admittedCond)
+
+	//reset Evicted condition if present.
```

Comment:
From https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties:

> Controllers should apply their conditions to a resource the first time they visit the resource, even if the status is Unknown.

Reply:
I opened this slack thread to learn more about the rationale for a required Reason https://kubernetes.slack.com/archives/C0EG7JC6T/p1682453786427459

But we will still need a reason for now. Perhaps it could be: NotAdmittedYet